transport-blockhound-tests/src/test/java/io/netty/util/internal/NettyBlockHoundIntegrationTest.java

```diff
@@ -127,6 +130,50 @@ private static void testEventExecutorTakeTask(EventExecutor eventExecutor) throw
         latch.await();
     }
 
+    @Test(timeout = 5000L)
+    public void testGlobalEventExecutorAddTask() throws Exception {
+        testEventExecutorAddTask(GlobalEventExecutor.INSTANCE, GlobalEventExecutor.class.getDeclaredField("taskQueue"));
+    }
+
+    @Test(timeout = 5000L)
+    public void testSingleThreadEventExecutorAddTask() throws Exception {
+        SingleThreadEventExecutor executor =
+                new SingleThreadEventExecutor(null, new DefaultThreadFactory("test"), true) {
+                    @Override
+                    protected void run() {
+                        while (!confirmShutdown()) {
+                            Runnable task = takeTask();
+                            if (task != null) {
+                                task.run();
+                            }
+                        }
+                    }
+                };
+        testEventExecutorAddTask(executor, SingleThreadEventExecutor.class.getDeclaredField("taskQueue"));
+    }
+
+    private void testEventExecutorAddTask(EventExecutor executor, Field taskQueueField) throws Exception {
+        // artificially acquire taskQueue.putLock
+        taskQueueField.setAccessible(true);
+        LinkedBlockingQueue<?> taskQueue = (LinkedBlockingQueue<?>) taskQueueField.get(executor);
+        Field putLockField = LinkedBlockingQueue.class.getDeclaredField("putLock");
```

Comment:
using reflection will not really work with Java9+. So I think we need to come up with something different here.

Reply:
That might be impossible I'm afraid, but I'll investigate. I think that's why 09d38c8 did not introduce any test for this.

Reply:
If thats the case then we may need to bite the bullet and just remove the test.

Reply:
I came up with a little something that works for `SingleThreadEventExecutor`, but not for `GlobalEventExecutor`. So I suggest that we leave `GlobalEventExecutor` untested. Let me know if it's looking better this way.